src/humanloop/decorators/prompt.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import typing
+import uuid
+from functools import wraps
+from typing import Any, Callable, Mapping, Optional, Sequence
+
+from opentelemetry.sdk.trace import Span
+from opentelemetry.trace import Tracer
+
+if typing.TYPE_CHECKING:
+    from humanloop import ToolFunctionParams
+from humanloop.decorators.helpers import args_to_inputs
+from humanloop.eval_utils import File
+from humanloop.otel import TRACE_FLOW_CONTEXT, FlowContext
+from humanloop.otel.constants import HL_FILE_KEY, HL_FILE_TYPE_KEY, HL_LOG_KEY, HL_PATH_KEY
+from humanloop.otel.helpers import write_to_opentelemetry_span
+from humanloop.types.model_endpoints import ModelEndpoints
+from humanloop.types.model_providers import ModelProviders
+from humanloop.types.prompt_kernel_request_stop import PromptKernelRequestStop
+from humanloop.types.prompt_kernel_request_template import PromptKernelRequestTemplate
+from humanloop.types.response_format import ResponseFormat
+
+logger = logging.getLogger("humanloop.sdk")
+
+
+def prompt(
+    opentelemetry_tracer: Tracer,
+    path: Optional[str] = None,
+    # TODO: Template can be a list of objects?
+    model: Optional[str] = None,
+    attributes: Optional[dict[str, Any]] = None,
+    endpoint: Optional[ModelEndpoints] = None,
+    template: Optional[PromptKernelRequestTemplate] = None,
+    provider: Optional[ModelProviders] = None,
+    max_tokens: Optional[int] = None,
+    stop: Optional[PromptKernelRequestStop] = None,
+    temperature: Optional[float] = None,
+    top_p: Optional[float] = None,
+    presence_penalty: Optional[float] = None,
+    frequency_penalty: Optional[float] = None,
+    other: Optional[dict[str, Optional[Any]]] = None,
+    seed: Optional[int] = None,
+    response_format: Optional[ResponseFormat] = None,
+    tools: Optional[Sequence["ToolFunctionParams"]] = None,
+):
+    def decorator(func: Callable):
+        prompt_kernel = {}
+
+        if temperature is not None:
+            if not 0 <= temperature < 1:
+                raise ValueError(f"{func.__name__}: Temperature parameter must be between 0 and 1")
+            prompt_kernel["temperature"] = temperature
+
+        if top_p is not None:
+            if not 0 <= top_p <= 1:
+                raise ValueError(f"{func.__name__}: Top-p parameter must be between 0 and 1")
+            prompt_kernel["top_p"] = top_p
+
+        if presence_penalty is not None:
+            if not -2 <= presence_penalty <= 2:
+                raise ValueError(f"{func.__name__}: Presence penalty parameter must be between -2 and 2")
+            prompt_kernel["presence_penalty"] = presence_penalty
+
+        if frequency_penalty is not None:
+            if not -2 <= frequency_penalty <= 2:
+                raise ValueError(f"{func.__name__}: Frequency penalty parameter must be between -2 and 2")
+            prompt_kernel["frequency_penalty"] = frequency_penalty
```

Comment:
this looks like it'll be a huge maintenance burden - can we not use our request models to validate this (or just not validate this)

Reply:
ack removed validation